### DISCUSSION TITLE
Repeatable option with csv arguments

### DISCUSSION THREAD

**Discussion by Fung-Chai** [pallets/click#2346] [Q&A]
How do I get a repeatable option with cvs argument to return a list of the individual values?  I think an example is needed to clarify what I mean.  I want `-n 1,2 -n 3 -n 4,5,6` to return `[1, 2, 3, 4, 5, 6]`.

In my code, I set multiple=True and type=subclass of ParamType that calls split(',').  But for the above command-line example, I got `[[1,2], [3], [4,5,6]]`

Any suggestion?  Thanks in advance.

**Comment by jules43** [comment]
I'm not sure there's a built in way to do this, could you just flatten the list of lists before you use it?

ie

flat_opt_list = [opt for opt_list in list_of_lists for opt in opt_list]
